controller/backup_controller.go

```diff
@@ -307,6 +307,20 @@ func (bc *BackupController) reconcile(backupName string) (err error) {
 			bc.eventRecorder.Eventf(backup, corev1.EventTypeWarning, string(backup.Status.State), "Failed backup %s has been deleted: %s", backup.Name, backup.Status.Error)
 		}
 
+		snapshot, err := bc.ds.GetSnapshot(backup.Spec.SnapshotName)
+		if err != nil {
+			if !apierrors.IsNotFound(err) {
+				return err
+			}
```

Comment:
Does it need to error out or just do best effort?

Reply:
Yes, we can do best effort for it. Thanks.